### ISSUE TITLE
API: read_csv parsing of quoted "NA" value as NaN

### ISSUE THREAD

**Issue by jorisvandenbossche** [pandas-dev/pandas#15669]
From https://github.com/pandas-dev/pandas/issues/10647#issuecomment-123715600

In context of discussion on the default parsing of 'NA' in a csv file as NaN, I don't think we can change the default of parsing `NA`, but we *may consider* parsing of `"NA"` as a string instead of NaN:

> Another change that we would maybe be more likely to consider, is the parsing of quoted values of NA (so only changing "NA" not be converted automatically, but leaving NA and alike converted to NaN as it is now).
Personally, I would even this consider as a bug that it treats "NA" and NA the same, as I would expect that quoted values should be left untouched. But I don't know how long this behaviour has been this way.

Although I am not sure how invasive such a change would be, and difficult to assess, so maybe this is not worth risking many breakages?

**Comment by BrenBarn** [comment]
The most pressing issue for me is that, as far as I can tell, there is no way at all to write a "NaN" string value to a CSV as such.  Obtaining such a value always requires fiddling with the NA settings when *readng* the file; there's nothing you can do when *writing* it to avoid the issue.  I don't think this is a good situation.

This is maybe related to a more general problem that the default settings of `read_csv` and `to_csv` are not aligned, so that doing a default `to_csv()` followed by a default `read_csv()` of the file you just wrote doesn't give you the same data.  (In particular, it will add extra columns for the index, because `to_csv` writes it but `read_csv` doesn't read it as an index.)

One possibility might be to have a CSV "dialect" called "pandas" or something.  This would just fix the various CSV options (including things like `index` that are not really part of CSV per se) in such a way that `to_csv` and `read_csv` become a lossless round-trippable pair if you use this dialect.

**Comment by HHest** [comment]
Please consider a corollary to differentiating quotes vs unquoted "NA" strings: treat columns specified with dtype=object/str as quoted as well.

Example:

pd.read_csv(StringIO('a,b\nNA,NA\n0123,0123'), dtype={'a': str})
Out[1]: 
      a      b
0   NaN    NaN
1  0123  123.0
Would love to see column 'a' showing "NA, 0123" as strings.

This is a very typical problem for us, because we have so many "exchange code" = "NA". Currently we need to use keep_default_na=False, and roll our own na_values, which is cumbersome and error prone. Thanks.

**Comment by jorisvandenbossche** [comment]
> Would love to see column 'a' showing "NA, 0123" as strings.

Yes, but I don't think that is something we can change. The idea I proposed above is only to change `"NA"` (so quoted) to be preserved as string.

Although the problem with that idea is that currently I don't think there is a way to write data like that with `to_csv`

> Perhaps the NaN vetting process could be controlled per column? Perhaps by allowing na_filter to take a dictionary of {col_name: boolean}.

That sounds like a good idea from a user point of view (don't know how difficult it would be implementation wise).

cc @gfyoung

**Comment by gfyoung** [comment]
`na_filter` can already accept a dictionary per the 0.21.0 docs.

**Comment by patricktokeeffe** [comment]
> In context of discussion on the default parsing of 'NA' in a csv file as NaN, I don't think we can change the default of parsing NA, but we may consider parsing of "NA" as a string instead of NaN:

fwiw, this may not be possible without changing the csv parser behavior: I use `read_csv` to import files where the NA value is literally `"NAN"` but I have to specify `NAN` to `na_values` parameter because double-quotes are automatically consumed.
